### DISCUSSION TITLE
Driver close after finishing with out .quit()

### DISCUSSION THREAD

**Discussion by ahmed-syntax** [seleniumbase/SeleniumBase#3105] [General]
`from seleniumbase import SB

with SB(uc=True) as sb:
    url = "https://gitlab.com/users/sign_in"
    sb.uc_open_with_reconnect(url, 4)`
    Sb.type('username', 'theusername')

How can I make the driver never close till i give it the permission for that .quit()

**Comment by mdmintz** [comment]
The context manager formats close the driver after the indented block ends. You could try the Driver() format: https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_23
Another example: https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_driver_manager.py
Add uc=True for UC Mode.

**Reply by ahmed-syntax** [reply]
I already tried this but i miss some options like set_attribute()

**Reply by mdmintz** [reply]
The driver has access to all raw Selenium methods, plus new methods that were added from SeleniumBase.

**Reply by ahmed-syntax** [reply]
```
from seleniumbase import Driver

driver = Driver(uc=True, incognito=True)
url = "example.com"
driver.uc_open_with_reconnect(url, 4)
driver.set_attribute("#username", "type", "button")
```

then why i get this:   AttributeError: 'Chrome' object has no attribute 'set_attribute'

**Reply by mdmintz** [reply]
The driver doesn’t have all the methods as SB. You may have to find the element first, and then set the attribute from there.

**Reply by ahmed-syntax** [reply]
Is there any way that i can use SB without the browser stops after finishing the code without using.sleep()

**Reply by mdmintz** [reply]
breakpoint() will pause your script until you tell it to continue.

**Reply by ahmed-syntax** [reply]
And if i want to exit sb.quit() not working how i can exit it ?

**Reply by mdmintz** [reply]
sb.driver.quit()

**Reply by ahmed-syntax** [reply]
Man you are a legend thanks for this amazing work keep going all the best ❤️
